### PR TITLE
(QTM-426) - Comparative test with current Dropdown x deprecatedDropdown

### PR DESCRIPTION
**Pull request by douglasjr97**
## Description
https://jirasoftware.catho.com.br/browse/QTM-426

## Review guide
- [ ] Unit tests (yarn test:components)
- [ ] Regression
- [ ] Code review
- [ ] Documentation

## Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile

##
## Review in aplications
1 - clone jobSearch project (https://github.com/catho/job-search) and import quantum from that branch in this application

....
- [ ] build quantum
- [ ] import quantum locally ex: 
``` yarn add ../quantum/dist```
- [ ] Replace original Dropdown with the new one (DropdownLight) and test at component (Apply-modal)
## 
2 - clone candidate-onboarding_register-candidate project ( https://github.com/catho/candidate-onboarding_register-candidate_module) and import quantum from that branch in this application

....
### Start the aplication in local environment with NPM

```
npm start
```

Using server-test

```
cd server-test
npm install
npm start
```

* Now just access: http://localhost:3000
- [ ] build quantum
- [ ] import quantum locally ex: 
``` yarn add ../quantum/dist```
- [ ] Replace original Dropdown with the new one (DropdownLight) and test at component [FinalStep.tsx] and [Cities.tsx] 
 https://seguro.catho.com.br/cadastro-candidato

### CODE REVIEW

**Review by MarcosViniciusPC — COMMENTED** [review]
When was it decided that DropdownLight should have items with images? I remember a conversation about this but I don't know what conclusion we reached. Another point, if these items are kept with images, I believe that the screenshots should have the list of items open, because the visual change that the DropdownLight will have will be in the list of items and not in the DropdownLight, do you agree?

---

We talked about this, and we think that be useful because the original dropdown, thats the temas using, has this image props.
To avoid a possible breaking change Because of that, we think its better to also have this prop. WDYT? @rapahaeru 

---

Oh, right! @MarcosViniciusPC

**Comment by MarcosViniciusPC** [comment]
>

**Review by rapahaeru — CHANGES_REQUESTED** [review]
The regression tests with the combobox opened is missing

File: components/DropdownLight/DropdownLight.unit.test.jsx [hunk 1]
```diff
@@ -225,4 +230,19 @@ describe('<DropdownLight />', () => {
...
+    render(<DropdownLight items={itemsWithImageMock} />);
+
+    const dropdown = screen.getByRole('button');
+    userEvent.click(dropdown);
+
+    const optionItem = screen.getByRole('option', {
+      name: 'Visa image description',
+    });
+
+    const selectionItemImage = optionItem.querySelector('img');
```
Comment (rapahaeru) [inline components/DropdownLight/DropdownLight.unit.test.jsx:244]: i think this querySelector is not necessary.
```
  it('should show if there is icon at item', () => {
    render(<DropdownLight items={itemsWithImageMock} />);

    const dropdown = screen.getByRole('button');
    userEvent.click(dropdown);

    const optionItem = screen.getByRole('option', {
      name: 'Visa image description',
    });

    expect(optionItem).toBeInTheDocument();
```

is there any reason for its use?

**Review by rapahaeru — COMMENTED** [review]
## Review guide
- [x] Unit tests (yarn test:components)
- [x] Regression
- [ ] Code review
- [x] Documentation

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [x] Chrome
  - [ ] Safari
  - [ ] Mobile

File: components/DropdownLight/DropdownLight.unit.test.jsx (hunk 1 above)
Comment (douglasjr97) [reply components/DropdownLight/DropdownLight.unit.test.jsx:244]: I understand that JsVanilla's native DOM should be avoided for testing. However, in image tests the querySelector would be the most correct, especially if in the future we want to test a specific image.

here are two articles that talk more about the subject:

https://robertmarshall.dev/blog/test-images-in-react-with-jest/#:~:text=The%20Jest%20test%20would%20be,%3E)%20const%20testImage%20%3D%20document

https://medium.com/@drake_beth/how-to-test-images-in-react-a70053b1634a

Comment (rapahaeru) [reply components/DropdownLight/DropdownLight.unit.test.jsx:244]: I agree, this logic sounds correct. :+1:

**Review by MarcosViniciusPC — CHANGES_REQUESTED** [review]
[] Code review
[x] Unit
[x] Regression
[] Documentation

Acessibility

[x] Screen reader
[x] keyboard navigate

 Layout review
 [x] Chrome
 [x] Safari
 [x] Mobile

[x] Job-search

With the exception of cases where the dropdown didn't use the "autocomplete" and "ignoreSpecialChars" props, the application worked perfectly. What are we going to do about it? I know this question was a question raised in the jira task description and I don't know what the conclusion of that was

[x] ssr-simulator

Adjust titles of examples from the documentation (storybook), as there are examples that should have texts separated by spaces but are together. Ex: "WithArrayOfString".

Missing id property in DropdownLight typescript.

There are two element(Button and SelectionList) using id in DropdownLight implementation and this is not correct

When focusing on a list element the styling is different between the old and new dropdown, is that expected?

**Perceptions about accessibility - screen reader**

- When focusing on the Dropdown Light button the text read is: "Open item list, popup button".
- When focusing on the old Dropdown button the text read is: "Fruits (dropdown label), combo box, collapsed, open menu, popup button".

My conclusions: The old one is much more descriptive than the new one because it reads the button's label, informs the type of element that will be opened, the state that the element is currently in (collapsed) and finally the button with the description of what action will occur when you press it.

- When opening Dropdown Light the text read is: "List with 5 items, lemon selected",
- When opening the old Dropdown the text read is: "cinco results... , use the appended down arrow key to navigate, press enter key to selecte".

The old dropdown is much more descriptive as it tells how to navigate and select an item in the list (despite being in English). Also, the new behavior says that the first item in the list is selected, but in the test done the item was just focused and these are different things.

My opinion is that we should do a task to adjust these points, what do you think?

In addition to the points mentioned, I find it interesting to centralize all the answers to the checklist requested in the task description in jira (https://jirasoftware.catho.com.br/browse/QTM-426) here in pr, what do you think?

**Review by rapahaeru — CHANGES_REQUESTED** [review]
when i change the Dropdown to the new version, after i select the option, the return of selected is the value instead the label
image

another example, step to step:
image

When selected
image

In the contract type Dropdown, the problem occurs in the layout too
image

All the points with original Dropdown

image

**Review by rapahaeru — APPROVED** [review]

**Pull request merged**